### DISCUSSION TITLE
Run vscode cppbuild tasks with git bash

### DISCUSSION THREAD

**Discussion by FieryRMS** [microsoft/vscode-cpptools#10477] [Q&A] [answered]
I am trying to make my `cppbuild` task on vscode to run on the git bash instead the windows cmd. Below is a minimum reproducible example for the task,
```json
{
    "type": "cppbuild",
    "label": "test",
    "command": "help",
    "args": []
}
```

My `terminal.external.windowsExec` is set to `C:\Program Files\Git\bin\bash.exe` and `terminal.integrated.defaultProfile.windows` is set to `Git Bash` as well, yet it still runs on the windows cmd. Changing `type` to `shell` does make it run on bash though, but I'd prefer to change the `cppbuild` default shell instead.

**Comment by FieryRMS** [comment, marked as answer]
I solved it myself, this was the process.

1. Open the folder containing all the extension code with vscode, for windows it's `%USERPROFILE%\.vscode\extensions`
1. Disable auto-updates. If you want to disable just for the C/C++ extension, follow [this tutorial](https://stackoverflow.com/questions/51965933/how-to-stop-a-single-vs-code-extension-from-auto-update).
1. `ctrl+shift+f` then paste this line, ```command = `cmd /c chcp 65001>nul && ${command}`;```
1. Open the file with that line, and comment it, under that line, paste,
`this.options.shell = 'bash';` or use the shell of your choice.
1. save and reload vscode, now it should use your preferred shell.

**Reply by fvydjt** [reply]
It works for me. By the way, the desired file is CPPTOOLS_EXTENSION_FOLDER\dist\main.js. For me, it's %USERPROFILE%\.vscode\extensions\ms-vscode.cpptools-1.14.5-win32-x64\dist\main.js and the modified command line number is 45926.
